### PR TITLE
Farming weight

### PR DESCRIPTION
**Pull request by metalcupcake5**
adds farming weight to farming section
image

also updates my front page message

### CODE REVIEW

File: src/lib.js [hunk 1]
```diff
@@ -3491,6 +3492,158 @@ async function getForge(userProfile) {
...
+
+  if (userProfile?.farming) {
+    let double_drops = userProfile.farming?.perks?.double_drops ?? 0;
+    output.bonuses.double_drops = {
+      double_drops: double_drops,
+      weight: double_drops * 2,
+    };
+    bonus += double_drops * 2;
+
+    let gold_medals = userProfile.farming?.total_badges?.gold ?? 0;
```
Comment (ptlthg) [inline src/lib.js]: Note that the gold medal counter the site has is bugged, so the weight total might differ slightly. #1853 fixes this.

File: src/lib.js [hunk 2]
```diff
@@ -3491,6 +3492,158 @@ async function getForge(userProfile) {
...
+        level: 0,
+        weight: 0,
+      },
+    },
+  };
+
+  const crops = {
+    CACTUS: {
+      name: "Cactus",
+      weight: 169_389.33,
```
Comment (ptlthg) [inline src/lib.js]: All the weight values should probably be moved into a constants file for easily adjusting them in the future.

File: views/stats.ejs [hunk 3]
```diff
@@ -1655,6 +1655,22 @@ const metaDescription = getMetaDescription()
             <% const farming = calculated.farming;
           if (farming.contests.attended_contests > 0) {%>
             <p class="stat-raw-values">
+              <span class="stat-name">Farming Weight: </span><span class="stat-value">
```
Comment (ptlthg) [inline views/stats.ejs]: Should this only be shown if the player has participated in a contest, as it is now? Their farming weight will surely be low if they haven't, but still. The same with pelts, actually.

File: src/lib.js [hunk 4]
```diff
@@ -3491,6 +3492,158 @@ async function getForge(userProfile) {
...
+    WHEAT: {
+      name: "Wheat",
+      weight: 100_000,
+    },
+  };
+
+  if (userProfile?.collections) {
+    let weight = 0;
+    for (const [name, crop] of Object.entries(crops)) {
+      let total = userProfile.collections[name].amount;
```
Comment (ptlthg) [inline src/lib.js]:
```suggestion
      let total = userProfile.collections[name]?.amount ?? 0;
```
Added a fallback to zero, as I'm not sure if locked collections would still appear in here.

File: src/lib.js (hunk 2 above)
Comment (MartinNemi03) [reply src/lib.js]: I would also consider moving farming weight calculation to `weight/farming-weight.js`

**Comment by ptlthg** [comment]
Before this can be merged, there's new weight values for the following after the Garden 1.5 update.

	pumpkin: 90_066.27,
	melon: 450_324.6,
	mushroom: 90_178.06,
	cocoa: 267_174.04,
	cactus: 177_254.45,
	
Axe crops were nerfed a tiny bit because of mossy fermento helm being better than lantern helm, and the Fungi Cutter was patched so mushroom weight is almost doubled now.

I do agree that this should all be moved into its own weight file like the other weight systems have, though.

**Edit:** Could also consider periodically fetching https://elitebot.dev/api/weights to get the current values, but they won't change often.

**Comment by metalcupcake5** [comment]
would it be possible to create an NPM package so its just a simple package update?

File: views/stats.ejs (hunk 3 above)
Comment (metalcupcake5) [reply views/stats.ejs]: added

File: src/lib.js (hunk 2 above)
Comment (metalcupcake5) [reply src/lib.js]: added

**Comment by ptlthg** [comment]
> would it be possible to create an NPM package so its just a simple package update?

I can definitely look into that for the future. I just have no free time right now, so it'll probably have to wait until summer.

This all looks good to me now though, thanks for adding everything!

Do I get a prestigious mention with the other weight systems in the about modal, though?
image

**Pull request merged**